addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3188,7 +3188,24 @@ export class OdooEditor extends EventTarget {
         this.deleteRange();
     }
     _onClipboardCopy(clipboardEvent) {
-        if (this.isSelectionInEditable()) {
+        const hasPartiallySelectedTable = hasTableSelection(this.editable) &&
+            ![...this.editable.querySelectorAll('.o_selected_table td')].every(td => (
+                !closestElement(td, 'table').classList.contains('o_selected_table') || // ignore nested
+                td.classList.contains('o_selected_td')
+            ));
```

Comment:
Each time I tried to review this PR I got stuck here and gave up. ^^' This time I'm going over it and leaving this message instead. I'm sorry, I tried, but that is impossible to read... ^^'

Reply:
Is this better ?
```js
const selectedTable = this.editable.querySelector('.o_selected_table');
const selectedTds = selectedTable &&
    [...selectedTables[0].querySelectorAll('td')].filter(td => !closestElement(td.parentElement, 'td')); // ignore nested
const hasPartiallySelectedTable = selectedTds && selectedTds.length && !selectedTds.every(td => td.classList.contains('o_selected_td'));
```

Reply:
I'm curious. Wouldn't `descendants(table, 'tr')` do what you want ? I suppose it stops when it finds a descendent and does not go further down looking for descendants in the descendants themselves, thus protecting you from nested tables ? If it works then it could also simplify the other code with the `tableClone`.

Reply:
`descendants` doesn't take a selector, which makes sense because that would be equivalent to `querySelectorAll`. Also anyway we don't want to stop when we found one because we want _all_ the top level tds.